core/server/data/migrations/versions/2.37/01-add-self-signup-and-from address-to-members-settings.js

```diff
@@ -0,0 +1,67 @@
+const _ = require('lodash');
+const Promise = require('bluebird');
+const common = require('../../../../lib/common');
+
+module.exports.config = {
+    transaction: true
+};
+
+module.exports.up = (options) => {
+    let localOptions = _.merge({
+        context: {internal: true}
+    }, options);
+    const settingsKey = 'members_subscription_settings';
+    return localOptions
+        .transacting('settings')
+        .then((response) => {
+            if (!response) {
+                common.logging.warn('Cannot find settings.');
+                return;
+            }
+            let subscriptionSettingsEntry = response.find((entry) => {
+                return entry.key === settingsKey;
+            });
+
+            if (!subscriptionSettingsEntry) {
+                common.logging.warn('Cannot find members subscription settings.');
+                return;
+            }
+            let subscriptionSettings = JSON.parse(subscriptionSettingsEntry.value);
+
+            let hasRequirePaymentProperty = Object.prototype.hasOwnProperty.call(subscriptionSettings, 'requirePaymentForSignup');
+            let hasSelfSignupProperty = Object.prototype.hasOwnProperty.call(subscriptionSettings, 'allowSelfSignup');
+            let hasFromAddressProperty = Object.prototype.hasOwnProperty.call(subscriptionSettings, 'fromAddress');
+
+            if (!hasFromAddressProperty) {
+                subscriptionSettings.fromAddress = 'noreply';
+            }
+
+            if (!hasRequirePaymentProperty && !hasSelfSignupProperty) {
```

Comment:
@rishabhgrg are the `requirePaymentForSignup` and `allowSelfSignup` properties in settings mutually exclusive? Meaning if one is set the other is not needed?

Reply:
disregard. saw the if's below :sweat_smile: